src/main/java/org/opensearch/agent/tools/SearchMonitorsTool.java

```diff
@@ -102,89 +98,73 @@ public <T> void run(Map<String, String> parameters, ActionListener<T> listener)
             ? Integer.parseInt(parameters.get("startIndex"))
             : 0;
 
-        // If a monitor ID is specified, all other params will be ignored. Simply return the monitor details based on that ID
-        // via the get monitor transport action
+        List<QueryBuilder> mustList = new ArrayList<QueryBuilder>();
         if (monitorId != null) {
-            GetMonitorRequest getMonitorRequest = new GetMonitorRequest(monitorId, 1L, RestRequest.Method.GET, null);
-            ActionListener<GetMonitorResponse> getMonitorListener = ActionListener.<GetMonitorResponse>wrap(response -> {
-                Monitor monitor = response.getMonitor();
-                processGetMonitorHit(monitor, listener);
-            }, e -> {
-                // System index isn't initialized by default, so ignore such errors. Alerting plugin does not return the
-                // standard IndexNotFoundException so we parse the message instead
-                if (e.getMessage().contains("Configured indices are not found")) {
-                    processGetMonitorHit(null, listener);
-                } else {
-                    log.error("Failed to get monitor.", e);
-                    listener.onFailure(e);
-                }
-            });
-            AlertingPluginInterface.INSTANCE.getMonitor((NodeClient) client, getMonitorRequest, getMonitorListener);
-        } else {
-            List<QueryBuilder> mustList = new ArrayList<QueryBuilder>();
-            if (monitorName != null) {
-                mustList.add(new TermQueryBuilder("monitor.name.keyword", monitorName));
-            }
-            if (monitorNamePattern != null) {
-                mustList.add(new WildcardQueryBuilder("monitor.name.keyword", monitorNamePattern));
-            }
-            if (enabled != null) {
-                mustList.add(new TermQueryBuilder("monitor.enabled", enabled));
+            mustList.add(new TermQueryBuilder("_id", monitorId));
+        }
+        if (monitorName != null) {
+            mustList.add(new TermQueryBuilder("monitor.name.keyword", monitorName));
+        }
+        if (monitorNamePattern != null) {
+            mustList.add(new WildcardQueryBuilder("monitor.name.keyword", monitorNamePattern));
+        }
+        if (enabled != null) {
+            mustList.add(new TermQueryBuilder("monitor.enabled", enabled));
+        }
+        if (hasTriggers != null) {
+            NestedQueryBuilder nestedTriggerQuery = new NestedQueryBuilder(
+                "monitor.triggers",
+                new ExistsQueryBuilder("monitor.triggers"),
+                ScoreMode.None
+            );
+
+            BoolQueryBuilder triggerQuery = new BoolQueryBuilder();
+            if (hasTriggers) {
+                triggerQuery.must(nestedTriggerQuery);
+            } else {
+                triggerQuery.mustNot(nestedTriggerQuery);
```

Comment:
If we are adding these filters to find the monitors, I think it would be beneficial to also search by monitor type as well.

Reply:
This PR isn't adding any filters, it's just shown as updated due to tabbing since the overall if/else was removed. Prefer to leave the tool with the current set of params and add/remove/tune later on as more testing is done and we capture the breadth of questions and functional responses.

Reply:
Especially for initial release, we need to be weary about how many knobs we allow the LLM to tune. Ideally we cover as many questions as possible in a confident/consistent/accurate way, and slowly adjust or expand over time after we have a good gauge of performance.

Reply:
Fair enough, we can set this as a further enhancement, down the road.